crates/extensions/tedge_uploader_ext/src/actor.rs

```diff
@@ -62,23 +63,28 @@ impl UploadResponse {
 
 pub type UploadResult = Result<UploadResponse, UploadError>;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct UploaderActor {
     config: ServerConfig,
+    identity: Option<Identity>,
 }
 
 impl UploaderActor {
-    pub fn new() -> Self {
-        UploaderActor::default()
+    pub fn new(identity: Option<Identity>) -> Self {
```

Comment:
That works fine for "real" optional values, like the back off parameters, where someone may or may not want to configure it. But as far as the code is concerned, this identity is required and comes directly from `tedge_config` (the value happens to be an `Option<Identity>`). If we made this a builder, it would lead to bugs (because people would forget to supply their uploader with an identity) and cause needless pain destructuring the Option value to `Some(identity)` on the caller's side (so they could call the method). This adds a little bit of noise to tests, but `None` isn't a lot of noise (particularly when we consider a test for this has to spin up a server).